### PR TITLE
ESC-581: pin dependencies to resolve CVEs for Citi escalation

### PR DESCRIPTION
**Pull request by janjwerner-confluent**
## Problem
Following up on https://confluentinc.atlassian.net/browse/ESC-581
this is plugging all the currently known CVEs for Citi connector ingestion. 
##### Does this solution apply anywhere else?
- [ ] yes
- [ ] no

##### Testing done:
- [ ] Unit tests
- [ ] Integration tests
- [ ] System tests
- [ ] Manual tests

### CODE REVIEW

File: kafka-connect-s3/pom.xml [hunk 1]
```diff
@@ -73,21 +73,6 @@
         <dependency>
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-java-sdk-s3</artifactId>
-            <exclusions>
-            <exclusion>
```
Comment (pbadani) [inline kafka-connect-s3/pom.xml:77]: @janjwerner-confluent - does bumping AWS version to 1.12.524 solve for this exclusion ( CVE-2020-28491 )?

File: pom.xml [hunk 2]
```diff
@@ -87,6 +89,17 @@
                 <artifactId>snakeyaml</artifactId>
                 <version>${snakeyaml.version}</version>
             </dependency>
+            <!-- band aid until the connector catches up with updated pom parent -->
+            <dependency>
```
Comment (pbadani) [inline pom.xml:93]: snappy-java dependency is resolved in this PR -
https://github.com/confluentinc/kafka-connect-storage-cloud/pull/669/files

File: pom.xml [hunk 3]
```diff
@@ -63,10 +63,12 @@
         <!-- temporary fix by pinning the version until we upgrade to a version of common that contains this or newer version.
             See https://github.com/confluentinc/common/pull/332 for details -->
         <dependency.check.version>6.1.6</dependency.check.version>
-        <hadoop.version>3.2.4</hadoop.version>
+        <hadoop.version>3.3.6</hadoop.version>
+        <jackson.version>2.15.2</jackson.version>
```
Comment (pbadani) [inline pom.xml:67]: jackson dependency is resolved in this PR - 
https://github.com/confluentinc/kafka-connect-storage-cloud/pull/669/files

File: kafka-connect-s3/pom.xml (hunk 1 above)
Comment (janjwerner-confluent) [reply kafka-connect-s3/pom.xml:77]: yes, please see: https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk-core/1.12.524
if you check https://twistlock.tools.confluent-internal.io/#!/monitor/vulnerabilities/images/ci?search=Confluent%20Public%20Repo%20PR%20builder%2Fkafka-connect-storage-cloud%2FPR-660
the version brought in is: 2.13.4 - /tmp/repo/kafka-connect-s3/target/kafka-connect-s3-10.0.22-SNAPSHOT-package/share/java/kafka-connect-s3/jackson-dataformat-cbor-2.13.4.jar

File: pom.xml (hunk 2 above)
Comment (janjwerner-confluent) [reply pom.xml:93]: it was done in parallel. please note update to jetty.

File: kafka-connect-s3/pom.xml (hunk 1 above)
Comment (pbadani) [reply kafka-connect-s3/pom.xml:77]: ok.

**Review by pbadani — APPROVED** [review]

**Pull request merged**